ff4s/src/main/scala/ff4s/Store.scala

```diff
@@ -34,53 +35,37 @@ trait Store[F[_], State, Action] {
 
 object Store {
 
-  def apply[F[_]: Concurrent, State, Action](
-      initialState: State
-  )(
-      makeDispatcher: SignallingRef[F, State] => Action => F[Unit]
+  def apply[F[_]: Concurrent, State, Action](init: State)(
+      update: Action => State => (State, F[Option[Action]])
   ): Resource[F, Store[F, State, Action]] = for {
+    actionQ <- Queue.unbounded[F, Action].toResource
 
-    state0 <- SignallingRef.of[F, State](initialState).toResource
+    effectQ <- Queue.unbounded[F, F[Option[Action]]].toResource
```

Comment:
You could also consider using an `fs2.concurrent.Channel`. I can't think of any meaningful differences between the two for your usecase.